tsconfig.json

```diff
@@ -1,6 +1,5 @@
 {
   "compilerOptions": {
-    "allowJs": true,
```

Comment:
With the recent typescript version there is no need at all to define declaration files for packages. There are totally optional now. See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html and the "Untyped Imports" section.

I don't think that the tests needs to be ported to typescript cause there irrelevant for the production build. Maybe the way to go would be two different `tsconfig.json` files one for production and one for building and enabling/disabling what is needed?